Automattic/jetpack
# Boost: Fix skip proxy origins to load css during critical CSS generation

While generating critical CSS, the CSS files need to be proxied through the plugin to avoid CORS restrictions. Previously the proxy was skipped if the server hostname matched the request, or if there was no hostname provided in the URL. ~~Now, it has a filter hook to allow more control.~~

**Correction**: As @davidlonjon later pointed out, we don't really need the filter to make this work. The real issue is, the port was not taking into consideration while comparing hostname.

#### Changes proposed in this Pull Request:
Improved hostname comparison to take port into account.

##### Reason behind:
While developing locally through docker, the docker container internally would understand the hostname as `localhost`. However, we can also change the external port mapping through `PORT_WORDPRESS` env variable. 
This leads to an issue. The external hostname and internal hostname are different so they are proxied while loading assets for critical CSS generation. Since the external URLs of the asset would contain the port in the request, it would be unreachable internally by the plugin.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Specify a value for `PORT_WORDPRESS` in **tools/docker/.env**. The value should be something other than 80(as that would previously break). [Read more](https://github.com/Automattic/jetpack/blob/master/tools/docker/README.md).
- Restart your jetpack development environment and make sure you can now access the the dev site at the new port i.e. localhost:<PORT>
- Try to regenerate CSS.
- Critical CSS should now generate without any issue.
